src/coreclr/jit/valuenum.cpp

```diff
@@ -9061,6 +9061,27 @@ void Compiler::fgValueNumberTree(GenTree* tree)
                 }
                 break;
 
+            case GT_SELECT:
+            {
+                GenTreeConditional* const conditional = tree->AsConditional();
+
+                ValueNumPair vnpExcSet = ValueNumStore::VNPForEmptyExcSet();
+
+                // Collect the exception sets from our operands
+                vnpExcSet = vnStore->VNPUnionExcSet(conditional->gtCond->gtVNPair, vnpExcSet);
+                vnpExcSet = vnStore->VNPUnionExcSet(conditional->gtOp1->gtVNPair, vnpExcSet);
+                vnpExcSet = vnStore->VNPUnionExcSet(conditional->gtOp2->gtVNPair, vnpExcSet);
+
+                // The normal value is a new unique VN.
```

Comment:
> The normal value is a new unique VN.

Why? I would think you can use the "implicit" func for select: `VNFunc(GT_SELECT)`.